### PR TITLE
fix(transaction): improve legacy gasPrice to EIP-1559 dynamic fee

### PR DESCRIPTION
**Pull request by gacevicljubisa**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description
Previously, when a legacy gasPrice was provided, both gasFeeCap and 
gasTipCap were set to gasPrice, causing excessive priority fees 
(gasPrice - baseFee overpayment). 

Now correctly maps:
- gasFeeCap = gasPrice (preserves user's total ceiling)
- gasTipCap = gasPrice - baseFee (actual realizable tip)

Also adds improved error handling for edge cases.

### CODE REVIEW

**Comment by gacevicljubisa** [comment]
Example how to test:
```
curl -XPOST -H "Gas-Price:6000000" localhost:1633/stamps/870000000/17
```

File: pkg/transaction/wrapped/fee.go [hunk 1]
```diff
@@ -20,20 +20,44 @@ var (
...
+			return new(big.Int).Set(gasPrice), new(big.Int).Set(gasPrice), nil
+		}
+
+		baseFee := latestBlockHeader.BaseFee
+		if gasPrice.Cmp(baseFee) < 0 {
+			return nil, nil, fmt.Errorf("specified gas price %s is below current base fee %s", gasPrice, baseFee)
+		}
+
+		// nominal tip = gasPrice - baseFee
+		gasTipCap := new(big.Int).Sub(gasPrice, baseFee)
```
Comment (martinconic) [inline pkg/transaction/wrapped/fee.go:45]: what if gasPrice - baseFee < baseFee ? Should we have max(gasPrice - baseFee, baseFee) ?

Comment (gacevicljubisa) [reply pkg/transaction/wrapped/fee.go:45]: When a user specifies a `gasPrice`, they're explicitly setting their maximum willingness to pay. The code validates that this price is at least the `baseFee`, and if it's not, returns an error.  In EIP-1559, you cannot pay less than the base fee. And the tip is the remainder after accounting for the mandatory base fee.

Comment (martinconic) [reply pkg/transaction/wrapped/fee.go:45]: Ok, but if gasFee is high and close to gasPrice, than the resulting gasTip is close to 0 and has no effect.

Comment (gacevicljubisa) [reply pkg/transaction/wrapped/fee.go:45]: Yes. But it is up to user to set GasPrice he will spend, and no more then this

**Review by martinconic — APPROVED** [review]
